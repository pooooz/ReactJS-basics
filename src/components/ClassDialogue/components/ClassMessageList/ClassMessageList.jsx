import React from 'react';
import styles from './ClassMessageList.module.scss'
import {ClassMessage} from "./components/ClassMessage/ClassMessage";

export class ClassMessageList extends React.Component {
  render() {
    return (
      <div className={styles.message__list}>
        {this.props.messages.map((message) => <ClassMessage message={message.text} author={message.author}/>)}
      </div>
    );
  }
}