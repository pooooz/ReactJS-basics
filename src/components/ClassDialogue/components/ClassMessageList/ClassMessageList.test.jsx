import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ClassMessageList } from './ClassMessageList';

describe('ClassMessageList', () => {
  it('Render component', () => {
    render(
      <ClassMessageList messages={[{ message: 'Render', author: 'Render' }]} />
    );
  });

  it('Render with snapshot', () => {
    const { asFragment } = render(
      <ClassMessageList messages={[{ message: 'Render', author: 'Render' }]} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render with text', () => {
    render(
      <ClassMessageList messages={[{ message: 'Render', author: 'Render' }]} />
    );
    expect(screen.getByText(/Author: <\w*>/));
  });
});
