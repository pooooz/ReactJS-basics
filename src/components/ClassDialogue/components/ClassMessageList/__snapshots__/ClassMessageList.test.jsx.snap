// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassMessageList Render with snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <p />
      <p
        data-text="Author: <Render>"
      >
        Author: &lt;Render&gt;
      </p>
    </div>
  </div>
</DocumentFragment>
`;
